cmd/state-svc/manager.go

```diff
@@ -19,15 +18,25 @@ import (
 )
 
 type serviceManager struct {
-	cfg    *config.Instance
-	lockFp string
+	cfg *config.Instance
 }
 
 func NewServiceManager(cfg *config.Instance) *serviceManager {
-	return &serviceManager{cfg, filepath.Join(cfg.ConfigPath(), "state-svc.lock")}
+	return &serviceManager{cfg}
 }
 
 func (s *serviceManager) Start(args ...string) error {
+	err := s.cfg.GetLock()
+	if err != nil {
+		return errs.Wrap(err, "Failed to get configuration lock")
+	}
+	defer s.cfg.ReleaseLock()
+
+	// Reload the configuration to ensure that we have the latest pid information
+	err = s.cfg.ReloadUnsafe()
```

Comment:
This feels unnecessary. We should be able to record that we are already locked, and as such don't lock again.

Reply:
I hear you, was thinking about that myself, but it is more difficult to do than it sounds, because the mutex also synchronizes behavior between threads inside the same process.  And when we have to check whether we are already locked, I don't know how to differentiate between the following two cases:

- The process holds the lock already and the call comes from the same thread => we can proceed
- The process holds the lock already, but the call comes from a different thread => we need to wait until the other thread releases the lock.

Reply:
I have a different idea to make this cleaner though.  Let me implement and push, so we can discuss it.

Reply:
This proposal would add a `SetSafe()` method to the config instance.  It is more explicitly targeted to the use case of manipulating values that depend on previous values.